WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsListFragment.kt

```diff
@@ -0,0 +1,102 @@
+package org.wordpress.android.ui.stats.refresh
+
+import android.arch.lifecycle.Observer
+import android.arch.lifecycle.ViewModelProvider
+import android.arch.lifecycle.ViewModelProviders
+import android.os.Bundle
+import android.os.Parcelable
+import android.support.v4.app.Fragment
+import android.support.v4.app.FragmentActivity
+import android.support.v7.widget.LinearLayoutManager
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import kotlinx.android.synthetic.main.stats_list_fragment.*
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.ui.stats.refresh.StatsListViewModel.StatsListType
+import org.wordpress.android.util.DisplayUtils
+import org.wordpress.android.widgets.RecyclerItemDecoration
+import javax.inject.Inject
+
+class StatsListFragment : Fragment() {
```

Comment:
I'd argue that, in essence, all of these lists are just a list of cards that show different things. Sure, what they show can be different, but that's up to the individual card/insight to know what data it represents and how it is show. The idea here is that the list management is pretty much the same in every fragment so it could be factored out to avoid duplicate code.

Reply:
When I think about it, I think you're right. Fragment and Adapter don't care about the data, only about the way they display UiModel and that should be the same for both list 👍